packages/create-llama/templates/components/engine/agent/index.py

```diff
@@ -0,0 +1,19 @@
+import logging
+import os
+import json
+
+from app.engine.constants import STORAGE_DIR
+from app.engine.context import create_service_context
+from app.engine.tools import ToolFactory
+from llama_index.agent import OpenAIAgent
+from llama_index import (
+    StorageContext,
+    load_index_from_storage,
+)
+
+
+def get_chat_engine():
+    service_context = create_service_context()
+    tools = ToolFactory.from_env().to_tool_list()
+    # TODO: get index by calling get_index and add query tool
```

Comment:
see https://docs.llamaindex.ai/en/stable/examples/agent/react_agent_with_query_engine.html#build-query-engine-tools

something like:
```
query_engine = get_index().as_query_engine()
query_engine_tool = QueryEngineTool.from_defaults(query_engine=query_engine)
tools.append(query_engine_tool)
if isinstance(llm, OpenAI) and is_function_calling_model(llm.model):
               return OpenAIAgent.from_tools(
                    tools=[query_engine_tool],
                    llm=llm,
                    **kwargs,
                )
else:
                return ReActAgent.from_tools(
                    tools=[query_engine_tool],
                    llm=llm,
                    **kwargs,
                )
 ```
 Ideally we can use https://github.com/run-llama/llama_index/pull/10452/files